sumologic/resource_sumologic_generic_polling_source.go

```diff
@@ -342,17 +374,26 @@ func getPollingThirdPartyPathAttributes(pollingResource []PollingResource) []map
 
 	for _, t := range pollingResource {
 		mapping := map[string]interface{}{
-			"type":                          t.Path.Type,
-			"bucket_name":                   t.Path.BucketName,
-			"path_expression":               t.Path.PathExpression,
-			"limit_to_regions":              t.Path.LimitToRegions,
-			"limit_to_namespaces":           t.Path.LimitToNamespaces,
-			"limit_to_services":             t.Path.LimitToServices,
-			"use_versioned_api":             t.Path.UseVersionedApi,
-			"custom_services":               flattenCustomServices(t.Path.CustomServices),
-			"tag_filters":                   flattenPollingTagFilters(t.Path.TagFilters),
-			"sns_topic_or_subscription_arn": flattenPollingSnsTopicOrSubscriptionArn(t.Path.SnsTopicOrSubscriptionArn),
+			"type":                t.Path.Type,
+			"bucket_name":         t.Path.BucketName,
+			"path_expression":     t.Path.PathExpression,
+			"limit_to_regions":    t.Path.LimitToRegions,
+			"limit_to_namespaces": t.Path.LimitToNamespaces,
+			"limit_to_services":   t.Path.LimitToServices,
+			"use_versioned_api":   t.Path.UseVersionedApi,
+			"custom_services":     flattenCustomServices(t.Path.CustomServices),
+			"tag_filters":         flattenPollingTagFilters(t.Path.TagFilters),
+			// "sns_topic_or_subscription_arn": flattenPollingSnsTopicOrSubscriptionArn(t.Path.SnsTopicOrSubscriptionArn),
+			"namespace":      t.Path.Namespace,
+			"event_hub_name": t.Path.EventHubName,
+			"consumer_group": t.Path.ConsumerGroup,
+			"region":         t.Path.Region,
+		}
+
+		if t.Path.SnsTopicOrSubscriptionArn != nil {
```

Comment:
can you give some context to the changes on `SnsTopicOrSubscriptionArn` field in this PR? I see changes in a few other places for this field but not sure why we need them here

Reply:
I think I added this while debugging an exception in the acc test which indicates no changes should occur after applying the terraform plan.  This turned out to be fixed by adding  
```
lifecycle {
   ignore_changes = [authentication[0].private_key]
} 
```
Also I tried to eliminate the field after running terraform plan. But seems we cannot avoid that even with current change. We can remove these changes. 

<img width="517" alt="Screenshot 2024-04-25 at 3 58 38 PM" src="https://github.com/SumoLogic/terraform-provider-sumologic/assets/124813412/db3d420a-1b8b-4841-ac3b-437623b100d2">